File: SDKs/netboxdemo.com/2.8/go/pkg/models/operations/dcimdevicetypescreate.go
package operations

import (
	"openapi/pkg/models/shared"
)

type DcimDeviceTypesCreateRequest struct {
	Request shared.WritableDeviceType `request:"mediaType=application/json"`
}

type DcimDeviceTypesCreateResponse struct {
	ContentType string
	DeviceType  *shared.DeviceType
	StatusCode  int64
}

File: SDKs/netboxdemo.com/2.8/typescript/src/sdk/models/operations/dcimdevicetypescreate.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class DcimDeviceTypesCreateRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.WritableDeviceTypeInput;
}


export class DcimDeviceTypesCreateResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  deviceType?: shared.DeviceType;

  @SpeakeasyMetadata()
  statusCode: number;
}
